.github/workflows/run-validations.yml

```diff
@@ -1,24 +1,49 @@
 name: Validations
 
-# Controls when the action will run. Workflow runs when manually triggered using the UI
-# or API.
-on: [push]
+on:
+  push:
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+defaults:
+  run:
+    shell: bash
 
 jobs:
-  build:
-      name: "Validate .pubnub.yml"
-      runs-on: ubuntu-latest
-      steps:
-        - uses: actions/checkout@v3
-        - name: Use Node.js
-          uses: actions/setup-node@v3
-          with:
-            node-version: '12.x'
-        - name: Install dependencies
-          run: |
-            npm install ajv@6.12.6
-            npm install yaml@1.10.0
-            npm install node-fetch@2.6.1
-            npm install chalk@2.4.2
-        - name: Validate
-          run: GITHUB_TOKEN=${{ secrets.GH_TOKEN }} node ./.github/workflows/validate-yml.js
+  repository-validators:
+    name: "Validate .pubnub.yml"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v3
+      - name: Checkout validator action
+        uses: actions/checkout@v3
+        with:
+          repository: pubnub/client-engineering-deployment-tools
+          ref: v1
+          token: ${{ secrets.GH_TOKEN }}
+          path: .github/.release/actions
+      - name: "Run '.pubnub.yml' file validation"
+        uses: ./.github/.release/actions/actions/validators/pubnub-yml
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+  build-validation:
+    name: Validate build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Validate clean build
+        if: ${{ github.event_name != 'pull_request' && github.ref == 'refs/heads/master' }}
+        run: |
+          ./gradlew build
```

Comment:
Why do we need this here? Why can't it be the part of tests? It's a waste of time IMHO. It'll do the tests and a little bit more

Reply:
Is there anything, that `gradle` will be able to reuse after tests with `gradlew build` command?
I'm trying to separate them contextually, This is workflow, in which various checks will be performed. In Swift we have checks on whether it compatible with dependency managers and they can handle it.

Reply:
@kleewho  Actually, there is no time will be wasted this way,. because it is in separate job, which runs in parallel, compared to when it was one of steps in `run-tests.yml` workflow.

Reply:
> Actually, there is no time will be wasted this way,. because it is in separate job, which runs in parallel,

Ok. CPU cycles wasted :D 

> Is there anything, that gradle will be able to reuse after tests 

There's probably a lot that could be reused but only if tests had chance to run before and put the results into the cache. Because they'll run in parallel it won't happen

Reply:
Ok, how critical it to be in same workflow as tests? 
Right now there is a win in overall duration of checks, because all will start in parallel jobs. No win in context of time in total (if we count each job and sum time for them).

I thought it will be great to have it contextually separated, because it is not really a test, but verification of whether build can be made or not.

If you think, that it should be in tests workflow, I'll move it back there.

Reply:
Sure. We can have this contextually separated